packages/components/src/internal/components/entities/utils.ts

```diff
@@ -205,3 +206,23 @@ export function updateCellKeySampleIdMap(
     Object.assign(updatedCellKeyMap, cellKeyChanges.toAddOrUpdate);
     return updatedCellKeyMap;
 }
+
+const PARENT_KEY_DIVIDER = '|';
+
+export function createEntityParentKey(schemaQuery: SchemaQuery, id?: string): string {
+    const keys = [schemaQuery.schemaName, schemaQuery.queryName];
+    if (id) {
+        keys.push(id);
+    }
+    return keys.join(PARENT_KEY_DIVIDER).toLowerCase();
+}
+
+export function parseEntityParentKey(parentKey: string): string[] {
+    if (!parentKey)
+        return [];
+
+    const allParts = parentKey.split(PARENT_KEY_DIVIDER);
+    const result = allParts.splice(0, 3);
```

Comment:
Shouldn't this remove 2 elements (`schema` and `query`), leaving behind everything else as the `value`?

A comment here about why the splice is needed seems worthwhile. Also jest tests.

Reply:
Comment and jest added.